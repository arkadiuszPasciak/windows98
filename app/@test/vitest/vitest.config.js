import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
