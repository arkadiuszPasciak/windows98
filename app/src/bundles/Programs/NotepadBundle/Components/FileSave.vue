<template>
  <UIModal
    class="FileSave"
    :title="t('FileBundle.title-save')"
    :resize-window="false"
    :move-window="false"
    width="320px"
    height="100px"
    :modal-state="modalState"
    @close-modal="$emit('closeModal')"
  >
    <div class="content">
      <UIInput
        id="file-save-name"
        v-model:modelValue="fileNameModelValue"
        class="input"
        :label-name="t('FileBundle.file-name')"
        label-position="left"
      />

      <UIButton
        data-test="file-save-button-save"
        @click="saveFile"
      >
        {{ t('FileBundle.save') }}
      </UIButton>

      <UISelect
        id="file-types"
        v-model:model-value="selectTextTypeModelValue"
        class="select"
        :options="MFilesTextTypes"
        :label-name="t('FileBundle.save-as-type')"
        label-position="left"
      />

      <UIButton
        data-test="file-save-button-cancel"
        @click="$emit('closeModal')"
      >
        {{ t('FileBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
import {
	EFileTextTypes,
	type TFileTextTypes,
} from "@APP/src/bundles/Programs/NotepadBundle/Supports/File.supports"
import { useFileManager } from "@APP/src/configs/app"
import type { IFilesTextTypes } from "@APP|Bundles/NotepadBundle/Supports/File.supports"
import UIButton from "@APP|Bundles/UIButtonBundle/Components/UIButton.vue"
import UIInput from "@APP|Bundles/UIInputBundle/Components/UIInput.vue"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import UISelect from "@APP|Bundles/UISelectBundle/Components/UISelect.vue"
import { ref } from "vue"
import { useI18n } from "vue-i18n"

const MFilesTextTypes = [
	{
		value: "text/txt;charset=utf-8;",
		name: "FileBundle.text-txt",
	},
	{
		value: "text/doc;charset=utf-8;",
		name: "FileBundle.text-doc",
	},
	{
		value: "text/csv;charset=utf-8;",
		name: "FileBundle.text-csv",
	},
	{
		value: "text/rtf;charset=utf-8;",
		name: "FileBundle.text-rtf",
	},
] as Array<IFilesTextTypes>

const props = defineProps({
	modalState: {
		type: Boolean,
		required: true,
	},
	contentValue: {
		type: String,
		required: true,
	},
})

defineEmits(["closeModal"])

const { t } = useI18n()
const fileManager = useFileManager()

const fileNameModelValue = ref<string>(t("FileBundle.untitled"))
const selectTextTypeModelValue = ref<TFileTextTypes>(MFilesTextTypes[0].value)

const saveFile = (): void => {
	fileManager.saveFile(
		props.contentValue,
		fileNameModelValue.value,
		EFileTextTypes.TXT,
	)
}
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/NotepadBundle/Styles/FileSave.styles.scss"
/>
