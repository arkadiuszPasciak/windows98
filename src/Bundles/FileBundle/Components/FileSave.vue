<template>
  <UIModal
    class="FileSave"
    :title="t('FileBundle.title-save')"
    :resize-window="false"
    :move-window="false"
    :width="320"
    :height="100"
    :modal-state="modalState"
    @close-modal="$emit('closeModal')"
  >
    <div class="content">
      <UIInput
        id="file-save-name"
        v-model:modelValue="fileNameModelValue"
        class="input"
        :label-name="t('FileBundle.file-name')"
        label-position="left"
      />

      <UIButton data-test="file-save-button-save" @click="saveFile">
        {{ t('FileBundle.save') }}
      </UIButton>

      <UISelect
        id="file-types"
        v-model:model-value="selectTextTypeModelValue"
        class="select"
        :options="MFilesTextTypes"
        data-test="file-save-button-cancel"
        :label-name="t('FileBundle.save-as-type')"
        label-position="left"
      />

      <UIButton @click="$emit('closeModal')">
        {{ t('FileBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'
  import UIInput from '@Bundles/UIBundle/Components/UIInput.vue'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import UISelect from '@Bundles/UIBundle/Components/UISelect.vue'
  import { MFilesTextTypes } from '@Bundles/FileBundle/Mocks/File.mocks'
  import { saveFileOnUserDisk } from '@Bundles/FileBundle/Services/File.services'
  import {
    EFileTextTypes,
    TFileTextTypes,
  } from '@Bundles/FileBundle/Supports/File.supports'

  const props = defineProps({
    modalState: {
      type: Boolean,
      required: true,
    },
    contentValue: {
      type: String,
      required: true,
    },
  })

  defineEmits(['closeModal'])

  const { t } = useI18n()

  const fileNameModelValue = ref(t('FileBundle.untitled')) as Ref<string>

  const selectTextTypeModelValue = ref(
    MFilesTextTypes[0].value,
  ) as Ref<TFileTextTypes>

  const saveFile = (): void => {
    saveFileOnUserDisk(
      props.contentValue,
      fileNameModelValue.value,
      EFileTextTypes.TXT,
    )
  }
</script>

<i18n src="@Bundles/FileBundle/Locales/File.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/FileBundle/Styles/FileSave.styles.scss';
</style>
