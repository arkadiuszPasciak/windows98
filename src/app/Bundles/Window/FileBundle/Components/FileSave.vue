<template>
  <UIModal
    class="FileSave"
    :title="t('FileBundle.title-save')"
    :resize-window="false"
    :move-window="false"
    width="320px"
    height="100px"
    :modal-state="modalState"
    @close-modal="$emit('closeModal')"
  >
    <div class="content">
      <UIInput
        id="file-save-name"
        v-model:modelValue="fileNameModelValue"
        class="input"
        :label-name="t('FileBundle.file-name')"
        label-position="left"
      />

      <UIButton data-test="file-save-button-save" @click="saveFile">
        {{ t('FileBundle.save') }}
      </UIButton>

      <UISelect
        id="file-types"
        v-model:model-value="selectTextTypeModelValue"
        class="select"
        :options="MFilesTextTypes"
        :label-name="t('FileBundle.save-as-type')"
        label-position="left"
      />

      <UIButton
        data-test="file-save-button-cancel"
        @click="$emit('closeModal')"
      >
        {{ t('FileBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIButton from '@APP|Bundles/UIButtonBundle/Components/UIButton.vue'
  import UIInput from '@APP|Bundles/UIInputBundle/Components/UIInput.vue'
  import UIModal from '@APP|Bundles/UIModalBundle/Components/UIModal.vue'
  import UISelect from '@APP|Bundles/UISelectBundle/Components/UISelect.vue'
  import { MFilesTextTypes } from '@APP|Bundles/FileBundle/Mocks/File.mocks'
  import { saveFileOnUserDisk } from '@APP|Bundles/FileBundle/Services/File.services'
  import {
    EFileTextTypes,
    TFileTextTypes,
  } from '@APP|Bundles/FileBundle/Supports/File.supports'

  const props = defineProps({
    modalState: {
      type: Boolean,
      required: true,
    },
    contentValue: {
      type: String,
      required: true,
    },
  })

  defineEmits(['closeModal'])

  const { t } = useI18n()

  const fileNameModelValue = ref(t('FileBundle.untitled')) as Ref<string>

  const selectTextTypeModelValue = ref(
    MFilesTextTypes[0].value,
  ) as Ref<TFileTextTypes>

  const saveFile = (): void => {
    saveFileOnUserDisk(
      props.contentValue,
      fileNameModelValue.value,
      EFileTextTypes.TXT,
    )
  }
</script>

<i18n src="@APP|Bundles/FileBundle/Locales/File.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/FileBundle/Styles/FileSave.styles.scss"
/>
