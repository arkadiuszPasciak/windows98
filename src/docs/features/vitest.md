# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
