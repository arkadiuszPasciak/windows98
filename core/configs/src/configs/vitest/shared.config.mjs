import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
