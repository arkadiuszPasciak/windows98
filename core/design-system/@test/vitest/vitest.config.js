import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
