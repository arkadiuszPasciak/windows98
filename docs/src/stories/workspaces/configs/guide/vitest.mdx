import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

