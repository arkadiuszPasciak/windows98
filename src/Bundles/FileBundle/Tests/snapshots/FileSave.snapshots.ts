import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import FileSave from '@Bundles/FileBundle/Components/FileSave.vue'

const defaultWrapper = mount(FileSave, {
  props: {
    modalState: true,
    contentValue: 'This is text to save',
  },
})

describe('[FileBundle]<Snapshots>(FileSave)', () => {
  it('renders correctly component', async () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
