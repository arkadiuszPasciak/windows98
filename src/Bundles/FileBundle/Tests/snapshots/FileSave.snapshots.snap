// Vitest Snapshot v1

exports[`[FileBundle]<Snapshots>(FileSave) > renders correctly component 1`] = `
<div
  class="UIModal FileSave"
  data-v-119c719c=""
  data-v-99a75817=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 320px; height: 100px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-v-99a75817=""
      >
        Save as
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    <!--v-if-->
    
    <div
      class="content"
      data-v-119c719c=""
    >
      <div
        class="UIInput label-position-left input"
        data-v-119c719c=""
        data-v-e394574e=""
      >
        <label
          class="label"
          data-v-e394574e=""
          for="file-save-name"
        >
          File name:
        </label>
        <input
          class="input"
          data-v-e394574e=""
          id="file-save-name"
          type="text"
        />
      </div>
      <button
        class="UIButton size-medium"
        data-v-119c719c=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        Save
        
        
        
      </button>
      <div
        class="UISelect label-position-left select"
        data-v-119c719c=""
        data-v-d3c68ad1=""
      >
        <label
          class="label"
          data-v-d3c68ad1=""
          for="file-types"
        >
          Save as type:
        </label>
        <select
          class="select"
          data-v-d3c68ad1=""
          id="file-types"
          name="file-types"
        >
          
          <option
            class="option"
            data-v-d3c68ad1=""
            value="text/txt;charset=utf-8;"
          >
            Text Document (*.txt)
          </option>
          <option
            class="option"
            data-v-d3c68ad1=""
            value="text/doc;charset=utf-8;"
          >
            Word Document (*.doc)
          </option>
          <option
            class="option"
            data-v-d3c68ad1=""
            value="text/csv;charset=utf-8;"
          >
            CSV Document (*.csv)
          </option>
          <option
            class="option"
            data-v-d3c68ad1=""
            value="text/rtf;charset=utf-8;"
          >
            Rich text format (*.rtf)
          </option>
          
        </select>
      </div>
      <button
        class="UIButton size-medium"
        data-v-119c719c=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        Cancel
        
        
        
      </button>
    </div>
    
  </div>
</div>
`;
